modules/imaging_browser/php/mrinavigation.class.inc

```diff
@@ -29,7 +29,7 @@ class MRINavigation
 {
     private array $_bits;
     private int $_currentListIndex;
-    private array $_FilteredSessionList;
+    private ?array $_FilteredSessionList;
```

Comment:
How does it get set to null? In the constructor right below this it's initialized as:

```
        $this->_FilteredSessionList = $_SESSION['State']
            ->getProperty('mriSessionsListed');
        if (!is_array($this->_FilteredSessionList)) {
            $this->_FilteredSessionList = [];
        }
```

Reply:
Here:
```
$this->_FilteredSessionList = $_SESSION['State']
            ->getProperty('mriSessionsListed');
```

the value returned by `$_SESSION['State']->getProperty('mriSessionsListed')` is null. The if case is not evaluated because a fatal is triggered, as the type of $this->_FilteredSessionList does not accept null value.

That could be fixed with something like:


```
$this->_FilteredSessionList = $_SESSION['State']
    ->getProperty('mriSessionsListed') ?? [];
```

Reply:
That makes sense. The check isn't done until after the assignment..

I think it would be better to not change the type, so that places dealing with the property don't need to worry about the distinction between null and empty array